test/test.ml

```diff
@@ -245,6 +245,49 @@ let test_presort () =
 
 (* *)
 
+let test_presort_and_is_in_log_xt () =
+  let n_incs = 50_000 and n_domains = 3 and n_locs = 12 in
+  let n_locs_half = n_locs asr 1 in
+
+  let barrier = Barrier.make n_domains in
+
+  let locs = Array.init n_locs (fun _ -> Loc.make 0) in
+
+  let in_place_shuffle array =
+    let n = Array.length array in
+    for i = 0 to n - 2 do
+      let j = Random.int (n - i) + i in
+      let t = array.(i) in
+      array.(i) <- array.(j);
+      array.(j) <- t
+    done
+  in
+
+  let thread () =
+    let locs = Array.copy locs in
+    Random.self_init ();
+    Barrier.await barrier;
+    for _ = 1 to n_incs do
+      in_place_shuffle locs;
+      let tx ~xt =
+        for i = 0 to n_locs_half - 1 do
+          Xt.incr ~xt locs.(i)
+        done;
+        assert (Xt.is_in_log ~xt locs.(Random.int n_locs_half));
+        assert (not (Xt.is_in_log ~xt locs.(n_locs_half)))
```

Comment:
The idea here is to exercise the `splay` implementation thoroughly.  The `locs` are shuffled, so the accesses are made in a random order.